app/Main.hs

```diff
@@ -108,6 +112,11 @@ optionsParser env = Options
        <*> switch
            (  long "no-inherit-env"
            <> help "don't merge the parent environment with the secrets file")
+       <*> (MilliSeconds <$> option auto
+               (  long "retry-base-delay"
```

Comment:
It would be a good idea to also put the time unit in the flag: somebody who encounters
```
vaultenv --retry-base-delay 100 --token ABC --secrets-file ~/channable/warpmachine/package/config/warpmachine.secrets  env
```
in a shell script or systemd unit file, has no clue whether those are 100 seconds, milliseconds, nanoseconds, or Martian days.
```
vaultenv --retry-base-delay-milliseconds 100 --token ABC --secrets-file ~/channable/warpmachine/package/config/warpmachine.secrets  env
```
on the other hand is self-documenting and leaves no room for ambiguity.

Reply:
I agree with @ruuda. I deeply dislike having no certainty about the units of values I have to pass to a program.

Reply:
Agreed! I'll do that